### PR TITLE
US173-T177: Switch order controller on frontend

### PR DESCRIPTION
**Pull request by evgenii-nikulin**
- Changed api call in frontend
- Changed pagination realization
- Added order builder in backend

### CODE REVIEW

File: src/Feature/Checkout/client/Integration/OrderHistory/sagas.ts [hunk 1]
```diff
@@ -30,15 +30,14 @@ export function* getOrderHistory(requestData: Action<OrderHistoryRequestPayload>
...
+    const { data, error }: Result<Commerce.Order[]> = yield call(Order.getOrders, firstPageNumber, ItemsPerPage);
 
     if (error) {
       return yield put(actions.GetOrderHistoryFailure(error.message || 'Error Occured'));
     }
 
-    const orders = data.orders as Commerce.OrderModel[];
-    const isLastPage = !orders || orders.length < ItemsPerPage;
-    yield put(actions.GetOrderHistorySuccess(orders, data.currentPageNumber, isLastPage));
+    const isLastPage = !data || data.length < ItemsPerPage;
```
Comment (yuya-yuliya) [inline src/Feature/Checkout/client/Integration/OrderHistory/sagas.ts:39]: Will we see "LOAD MORE" button if count of orders on last page equal to items per page?

File: src/Foundation/Commerce/code/Builders/Order/OrderBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        public Order Build(Sitecore.Commerce.Entities.Orders.Order source)
+        {
+            var result =
+                this.orderMapper.Map<Models.Entities.Cart.Cart, Order>(this.cartBuilder.Build(source));
+
+            result.OrderID = source.OrderID;
+            result.OrderDate = source.OrderDate;
+            result.Status = source.Status;
+            result.TrackingNumber = source.TrackingNumber;
+            result.IsOfflineOrder = source.IsOfflineOrder;
```
Comment (yuya-yuliya) [inline src/Foundation/Commerce/code/Builders/Order/OrderBuilder.cs:53]: We can use AutoMapper for this

File: src/Wooli.sln [hunk 3]
```diff
@@ -0,0 +1,244 @@
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 16
```
Comment (yuya-yuliya) [inline src/Wooli.sln:2]: Delete this file, because we have new `.sln` file with HCA name

**Review by yuya-yuliya — APPROVED** [review]

File: src/Feature/Checkout/client/Integration/OrderHistory/sagas.ts [hunk 4]
```diff
@@ -30,15 +30,14 @@ export function* getOrderHistory(requestData: Action<OrderHistoryRequestPayload>
   try {
     yield put(actions.GetOrderHistoryRequest());
     const firstPageNumber = 0;
-    const { data, error }: Result<Commerce.OrderHistoryResultModel> = yield call(Order.getOrders, firstPageNumber, ItemsPerPage);
+    const { data, error }: Result<Commerce.Order[]> = yield call(Order.getOrders, firstPageNumber, ItemsPerPage);
```
Comment (larionovadaria) [inline src/Feature/Checkout/client/Integration/OrderHistory/sagas.ts:33]:
```
export interface OrderHistoryResultModel {
        currentPageNumber: number;
        orders: CartModel[];
    }
```
1. Don't we actually need `currentPageNumber` anymore?
2. Now orders are also cart models?

File: src/Foundation/Commerce/client/dataModel.Generated.ts [hunk 5]
```diff
@@ -14,326 +14,354 @@
...
+}
+export interface OrderHistoryResultModel {
+    currentPageNumber: number;
+    orders: CartModel[];
+}
+export interface UserLoginModel {
+    email: string;
+    password: string;
+}
+export interface Order extends Cart {
```
Comment (larionovadaria) [inline src/Foundation/Commerce/client/dataModel.Generated.ts:324]: Why does Order extend Cart? This is 2 different entities, in my opinion. 
Order could have a link to cart (maybe) but why extend?

File: src/Foundation/Commerce/code/Models/Entities/Order/Order.cs [hunk 6]
```diff
@@ -17,16 +17,20 @@ namespace HCA.Foundation.Commerce.Models.Entities.Order
...
     using System.Diagnostics.CodeAnalysis;
 
+    using Cart;
+
     using TypeLite;
 
     [ExcludeFromCodeCoverage]
     [TsClass]
-    public class Order
+    public class Order : Cart
```
Comment (larionovadaria) [inline src/Foundation/Commerce/code/Models/Entities/Order/Order.cs:26]: ` public class Order : Cart` 
Why?

**Review by ghost — CHANGES_REQUESTED** [review]
The main question is about inheritance between order and cart.

File: src/Foundation/Commerce/client/dataModel.Generated.ts (hunk 5 above)
Comment (evgenii-nikulin) [reply src/Foundation/Commerce/client/dataModel.Generated.ts:324]: Sitecore entities `Order` extends `Cart`. Old structure of this project do this too. Besides many of `Cart` fields uses in frontend.

File: src/Feature/Checkout/client/Integration/OrderHistory/sagas.ts (hunk 4 above)
Comment (evgenii-nikulin) [reply src/Feature/Checkout/client/Integration/OrderHistory/sagas.ts:33]: 1. Yes, I don't know why in previous version we has it. It doesn't use at all, just passed through without changes. And actually track of which current page now is not backend task.
2. Yes, with some additional fields.

**Pull request merged**